### PR TITLE
Added configuration parameter lockDir to specify where proto.lock lives

### PR DESCRIPTION
**Pull request by seime**
Passing `<option>--lockdir=somewhere_else</option>` does not work, the plugin always checks the proto root directory for `proto.lock` .

This PR adds a new configuration parameter `lockDir`

### CODE REVIEW

File: src/main/java/com/salesforce/servicelibs/BackwardsCompatibilityCheckMojo.java [hunk 1]
```diff
@@ -189,15 +195,33 @@ public void execute() throws MojoExecutionException, MojoFailureException {
                 options = "";
             }
 
-            Path lockFile = Paths.get(protoSourceRoot, "proto.lock");
+            if (options.toUpperCase().contains("--LOCKDIR")) {
```
Comment (rmichela) [inline src/main/java/com/salesforce/servicelibs/BackwardsCompatibilityCheckMojo.java]: Move this new code section to a private method that returns the `lockDir`.  It's not your fault that `execute()` is getting really hairy, but it's time to start breaking it into methods.

File: src/main/java/com/salesforce/servicelibs/BackwardsCompatibilityCheckMojo.java [hunk 2]
```diff
@@ -189,15 +195,33 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
+                throw new MojoFailureException("lockDir location must be specified on the plugin and not as "
+                        + "an option passed to protolock command");
+            }
+
+            // Build option for lock file location
+            String _lockDir = null;
```
Comment (rmichela) [inline src/main/java/com/salesforce/servicelibs/BackwardsCompatibilityCheckMojo.java]: nit: There has to be a better variable name than `_lockDir`.

**Review by rmichela — CHANGES_REQUESTED** [review]
I'm curious, what prompted the need for this change?

**Comment by seime** [comment]
> I'm curious, what prompted the need for this change?

We generate some proto descriptors from xml schema, and as with all generated sources/resources we place them in the (maven) target folder before packaging and deploying them as jar files.

The plugin was hardcoded to always look for proto.lock inside the proto source folder, and if not found it would just renitialize a new proto.lock there each time. Even if --lockdir would be specified the maven plugin wouldn't honour that and just renitialize a new proto.lock inside the temp folder. This caused changes silently to be accepted (except for a log statement saying that it was initialized)

**Review by rmichela — APPROVED** [review]
Thanks for the extra refactoring!

**Comment by rmichela** [comment]
Released in [1.0.6](https://github.com/salesforce/proto-backwards-compat-maven-plugin/releases/tag/v1.0.6)

**Pull request merged**